numpy/core/src/multiarray/vec_string.c

```diff
@@ -0,0 +1,368 @@
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _MULTIARRAYMODULE
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "numpy/arrayobject.h"
+#include "dtypemeta.h"
+
+#include "stringlib/stringlib.h"
+#include "stringlib/length.h"
+#include "stringlib/isalpha.h"
+#include "stringlib/add.h"
+#include "stringlib/undef.h"
+
+#include "stringlib/ucs4lib.h"
+#include "stringlib/length.h"
+#include "stringlib/isalpha.h"
+#include "stringlib/add.h"
+#include "stringlib/undef.h"
+
+typedef int (*_vec_string_fast_op)(PyArrayIterObject **, PyArrayIterObject *);
+
+typedef struct {
+    const char *name;
+    _vec_string_fast_op unicode_func;
+    _vec_string_fast_op bytes_func;
+    int with_args;
+} _vec_string_named_fast_op;
+
+// This static array contains all of the np.char functions that support
+// fast ops, according to the length of the method name as a string.
+// For example, __add__ and isalpha are at index seven, cause they're
+// seven characters long.
```

Comment:
In some places what we use is a switch statement based on the first character (and or length?) and only then match I think.  But not sure.  Feels a bit oversizes on first sight, but I am OK with it.